### PR TITLE
ip-address-expansion example

### PR DESCRIPTION
**Pull request by elv1s**
example iControlLX extension for using the api-services-gateway container

### CODE REVIEW

**Comment by dajiabao** [comment]
ship it!

File: container/ip-address-expansion/GUIsrc/presentation/application.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import { Router, hashHistory } from 'react-router';
+import { syncHistoryWithStore } from 'react-router-redux';
+import store from './stores/configStore';
+import routes from './routes';
+let my_helper = require('./helpers/my_helper.js');
+
+export const history = syncHistoryWithStore(hashHistory, store);
+//let clientId = store.getState().iapp.get('clientId');
```
Comment (MrAshLinux) [inline container/ip-address-expansion/GUIsrc/presentation/application.js]: Remove commented line

File: container/ip-address-expansion/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM f5devcentral/f5-api-services-gateway
```
Comment (manjubhat) [inline container/ip-address-expansion/Dockerfile]: Are we tagging the version ? May be use specific version here ?

File: container/ip-address-expansion/GUIsrc/presentation/constants.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  maxErrorCount: 10,
+  appState: {
+    ONLINE: 'ONLINE',
+    OFFLINE: 'OFFLINE',
+    DISABLED: 'DISABLED',
+    ERROR: 'ERROR',
+    TRANSITION: 'TRANSITION',
+    UNKNOWN: 'UNKNOWN',
+  },
+  blockState: {
```
Comment (manjubhat) [inline container/ip-address-expansion/GUIsrc/presentation/constants.js]: I think iapp-address-expansion is a iControl LX and not using block state machine.. Probably Alex copied this from other projects

File: container/ip-address-expansion/nodejs/appWorker.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
+/*
+  Copyright (c) 2017, F5 Networks, Inc.
```
Comment (manjubhat) [inline container/ip-address-expansion/nodejs/appWorker.js:2]: We are not using same license on GUI code ? Is that okay ?

File: container/ip-address-expansion/presentation/index.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
```
Comment (manjubhat) [inline container/ip-address-expansion/presentation/index.html:1]: Seemed like two copies of same file exists in this pull request

File: container/ip-address-expansion/Dockerfile (hunk 2 above)
Comment (elv1s) [reply container/ip-address-expansion/Dockerfile]: good idea, we are tagging versions. I'll add one

File: container/ip-address-expansion/GUIsrc/presentation/constants.js (hunk 3 above)
Comment (elv1s) [reply container/ip-address-expansion/GUIsrc/presentation/constants.js]: yep, cleaned it up a bit

File: container/ip-address-expansion/nodejs/appWorker.js (hunk 4 above)
Comment (elv1s) [reply container/ip-address-expansion/nodejs/appWorker.js:2]: added f5 copyright, and apache 2.0 license to presentation source files.

File: container/ip-address-expansion/presentation/index.html (hunk 5 above)
Comment (elv1s) [reply container/ip-address-expansion/presentation/index.html:1]: yes, one is in the "GUIsrc" the other is the build output (We probably shouldn't check in the build output)

**Review by atenenb — APPROVED** [review]
Ship it!

**Pull request merged**